src/main/java/io/aiven/kafka/connect/opensearch/OpensearchSinkConnectorConfig.java

```diff
@@ -397,13 +404,38 @@ private static void addConversionConfigs(final ConfigDef configDef) {
                 group,
                 ++order,
                 Width.SHORT,
-                "Behavior on document's version conflict (optimistic locking)");
+                "Behavior on document's version conflict (optimistic locking)"
+        ).define(
+                ERRORS_TOLERANCE_CONFIG,
+                Type.STRING,
+                BulkProcessor.ErrorToleranceType.DEFAULT.toString(),
+                BulkProcessor.ErrorToleranceType.VALIDATOR,
+                Importance.LOW,
+                ERRORS_TOLERANCE_DOC,
+                group,
+                ++order,
+                Width.SHORT,
+                "Error Tolerance");
     }
 
     public static final ConfigDef CONFIG = baseConfigDef();
 
     public OpensearchSinkConnectorConfig(final Map<String, String> props) {
         super(CONFIG, props);
+        validateDeadLetterQueueConfig(props);
+    }
+
+    private void validateDeadLetterQueueConfig(final Map<String, String> props) {
+        final String dqlTopicStr = props.get(DLQ_TOPIC_NAME_CONFIG);
+        final boolean dlqIsConfigured = dqlTopicStr != null && !dqlTopicStr.isEmpty();
```

Comment:
This doesn't cover all cases. The connector may be instantiated with a non-null `ErrantRecordReporter` even if it isn't configured to use a DLQ topic (see [here](https://github.com/apache/kafka/blob/6e7e2e08a982c6667476a932e2571f8ef4b70dc3/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java#L984-L986), [here](https://github.com/apache/kafka/blob/trunk/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/SinkConnectorConfig.java#L167-L170), and [here](https://github.com/apache/kafka/blob/trunk/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/ConnectorConfig.java#L262-L264)).

I don't think we should rely on parsing user-supplied configurations that are defined outside of the connector (i.e., in the Connect runtime) to validate whether a reporter should be required or not.

Instead, high-level, I'm thinking we might do this:
- Add a `requiresErrantRecordReporter` method (feel free to rename) to the `OpensearchSinkConnectorConfig` class that returns whether or not the connector requires an `ErrantRecordReporter` instance to function properly
- In the `OpensearchSinkTask` class, check whether an `ErrantRecordReporter` instance is required (using the method above); if it is, and if [`getErrantRecordReporter`](https://github.com/Natural-Intelligence/opensearch-connector-for-apache-kafka/blob/e23ff3c33e5fd5c5014ecae0a90de3d8372f5723/src/main/java/io/aiven/kafka/connect/opensearch/OpensearchSinkTask.java#L88) returns `null`, throw an exception explaining the problem to the user

This way, we don't have to redefine any runtime-defined properties (like `errors.tolerance`), and we cover all our bases for what may cause the connector to be instantiated with a non-null `ErrantRecordReporter`, including any future changes to that API like allowing developers to define their own pluggable error-handling logic.

Does that seem reasonable?